### ISSUE TITLE
Ranger changes the terminal colorscheme after a file was opened and then closed.

### ISSUE THREAD

**Issue by Kharacternyk** [ranger/ranger#1983]
#### Runtime Environment
Terminal emulator: VTE\* and Alacritty are affected, haven't tested the others
ranger version: ranger 1.9.3
Python version: 3.8.3 (default, May 17 2020, 18:15:42) [GCC 10.1.0]
Locale: en_US.UTF-8

#### Current Behavior
Ranger changes the color scheme of the terminal it runs within after a file was opened and then closed.

#### Expected Behavior
The colorscheme is left as it is.

#### Steps to reproduce
1. Use escape sequences of form `\033]4;$COLOR_ID;$HEX_COLOR\007`. They change the colorscheme and are used, for example, by pywal.
2. Enter ranger. 
3. Open a file in ranger.
4. Close the file and return to ranger. The colors are changed (the bg and fg are not though) and look like ones from the Linux tty.

#### Screenshots
Before a file is opened, colorscheme gruvbox:
![1590944367_win_screenshot](https://user-images.githubusercontent.com/43315801/83358057-6c34ee80-a379-11ea-9be4-cd76391359b1.png)
After:
![1590944374_win_screenshot](https://user-images.githubusercontent.com/43315801/83358064-78b94700-a379-11ea-958f-470e4e8f3d06.png)

**Comment by Kharacternyk** [comment]
Just tested the Linux tty, the bug is present there. The escape sequences are different though, the usual ones don't work there.

**Comment by Kharacternyk** [comment]
Vifm is also affected.

**Comment by Kharacternyk** [comment]
Lf is **not** affected. I'm not sure what it does differently from ranger and vifm.

**Comment by mehalter** [comment]
I get the same behavior using Kitty terminal emulator

**Comment by toonn** [comment]
Do the colors return to those before issuing the escape sequences, be that by pywal or something else?

**Comment by Kharacternyk** [comment]
> Do the colors return to those before issuing the escape sequences, be that by pywal or something else?

Yes. I changed the default colorscheme of my terminal and ranger indeed resets to the new colorscheme.

**Comment by toonn** [comment]
Then I'm not sure this can be fixed. Programs might mess with colors so we have to reset the theme.
The only option I see is ranger storing what all the colors are set to and restoring to that rather than having the terminal do so. Maybe look into whether that can be implemented as a plugin.

**Comment by Kharacternyk** [comment]
Is there a possibility of having a config option for those who don't use programs that might mess with colors?

**Comment by toonn** [comment]
No, I can only see that leading to yet harder to debug issues related to colors.

**Comment by j-n-f** [comment]
@Kharacternyk : what is your `TERM` variable set to? I have the exact same problem in ranger (as well as ncmpcpp, another ncurses app). I can solve this issue for all programs exhibiting the issue by explicitly setting `TERM=xterm` (rather than `xterm-256color` as it's usually set to).

I'm also using Alacritty.

I just tested it out with urxvt (using `TERM=xterm` and `TERM=xterm-256color`) and the issue doesn't show up. Seems like this is unique to Alacritty (and Kitty as @mehalter observes).

Looks like the issue have been noticed but not resolved:

https://github.com/alacritty/alacritty/issues/2052

This will probably end up being a bug in Alacritty (as urxvt doesn't have the issue, and AFAIK ranger and ncmpcpp are using different libraries for ncurses).

**Comment by Kharacternyk** [comment]
The bug is present in VTE based terminals and the Linux TTY, too. I doubt that it's somehow specific to Alacritty or Kitty. A possible explanation is that RXVT doesn't support the escape sequence needed to reset the colors.

I don't think changing `$TERM` is a reliable solution. If I'm not mistaken, it'll forbid apps to use more than 16 colors.

**Comment by wandbrandon** [comment]
This effects ncmpcpp as well. setting the TERM variable does indeed fix it, but is obviously not a reliable solution. I could not reproduce it in Kitty, it works fine.

**Comment by carywreams** [comment]
### Same effect simply entering shell

FWIW, for others who end up here as I did, ...I'm seeing the same issue running ranger 1.9.3 in gnome-terminal on pop-os 2204, using bash shell. I'm also attempting to use base16 to apply consistent color palettes across bash, ranger, and nvim to ease eye strain.

+ enter ranger
+ shift-S to enter shell
+ exit/ctrl-D to leave shell (this is the point at which colors change in gnome-terminal)
+ Q to leave ranger (colors still changed)

I have changed .bashrc, to ``export TERM="xterm"`` as [mentioned here](https://github.com/ranger/ranger/issues/1983#issuecomment-668191499) understanding there are cautions to implementing this work-around. However, using the base16 approach may work in my favor here...
